Zend/zend_vm_def.h

```diff
@@ -5481,14 +5481,15 @@ ZEND_VM_HANDLER(181, ZEND_FETCH_CLASS_CONSTANT, VAR|CONST|UNUSED|CLASS_FETCH, CO
 				ZVAL_UNDEF(EX_VAR(opline->result.var));
 				HANDLE_EXCEPTION();
 			}
-			value = &c->value;
-			if (Z_TYPE_P(value) == IS_CONSTANT_AST) {
-				zval_update_constant_ex(value, c->ce);
-				if (UNEXPECTED(EG(exception) != NULL)) {
+			if (UNEXPECTED(!(c->ce->ce_flags & ZEND_ACC_CONSTANTS_UPDATED))) {
+				if (UNEXPECTED(zend_update_class_constants(c->ce)) != SUCCESS) {
```

Comment:
We need to update the entire class constants' table for some edge cases when doing nested fetching (i. e. https://pastebin.com/pimJvY8m; cc @nikic)